main/kotlin/com/example/people_managment_system/service/SystemService.kt

```diff
@@ -0,0 +1,177 @@
+package com.example.people_managment_system.service
+
+import com.example.people_managment_system.dtos.EmployeeDto
+import com.example.people_managment_system.model.Employee
+import com.example.people_managment_system.repositories.CompanyRepository
+import com.example.people_managment_system.repositories.EmployeeRepository
+import com.example.people_managment_system.repositories.ManagerRepository
+import com.example.people_managment_system.repositories.TeamRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.stereotype.Service
+
+@Service
+
+class SystemService(
+    @Autowired  val employeeRepository: EmployeeRepository,
+                val companyRepository: CompanyRepository,
+                val teamRepository: TeamRepository,
+                val managerRepository: ManagerRepository
+    )
+    {
+
+//  Retrieves an employee by their ID or throws an exception if the employeeID doesn't exist. //
+
+    fun retrieveEmployeeById(employeeID: Int): EmployeeDto {
+        var employeeFound: Employee? = null
+        for (item in employeeRepository.findAll()) {
+            if (item.employeeID == employeeID) {
+                employeeFound = item
+            }
+        }
+        return employeeFound?.let { mapEmployeeEntityToEmployeeDto(it) }
+            ?: throw NoSuchElementException("Could not find an employee with that ID")
+    }
+
+/*  Retrieves all the employees in a company by taking in the company name as a path variable and
+    returning a list of the employees.  Returns an empty list if no employees are found in the company,
+    or if the company name is invalid. */
+
+    fun retrieveEmployeesInACompany(companyName: String): MutableList<EmployeeDto> {
+    val employeesFound = mutableListOf<EmployeeDto>()
+    for (item in employeeRepository.findAll()) {

```

Comment:
This is a bonus, but you can take a look into high order functions, basically the modern way to program: https://kotlinlang.org/docs/lambdas.html